File: loop_transformations/sources/stripe.1.c
/*
* @@name:	stripe.1
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_6.0
*/
void func(int A[12][6])
{
  #pragma omp stripe sizes(4)
  for (int i = 0; i < 12; ++i)
    for (int j = 0; j < 5; ++j)
      A[i][j+1] = A[i][j]/32;
}

void func_equivalent(int A[12][6])
{
  for (int i1 = 0; i1 < 4; i1+=1)
    for (int i2 = i1; i2 < 12; i2+=4)
      for (int j = 0; j < 5; ++j)
        A[i2][j+1] = A[i2][j]/32;
}

File: loop_transformations/sources/stripe.1.f90
! @@name:	stripe.1
! @@type:	F-free
! @@operation:	compile
! @@expect:	success
! @@version:	omp_6.0
subroutine func(A)
  implicit none
  integer :: A(6,12)
  integer :: i, j

  !$omp stripe sizes(4)
  do i = 1, 12
    do j = 1, 5
      A(j+1,i) = A(j,i)/32
    end do
  end do
end subroutine

subroutine func_equivalent(A)
  implicit none
  integer :: A(6,12)
  integer :: i1, i2, j

  do i1 = 1, 4
    do i2 = i1, 12, 4
      do j = 1, 5
        A(j+1,i2) = A(j,i2)/32
      end do
    end do
  end do
end subroutine
